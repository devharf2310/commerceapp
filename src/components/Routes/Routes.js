import React from "react";
import { Route, Switch, Redirect } from 'react-router-dom';

import Home from '../../pages/Home/Home';
import Contacts from '../../pages/Contacts/Contacts';
import Clients from '../../pages/Clients/Clients';
import Products from '../../pages/Products/Products';

const Routes = ({location}) => (

    <Switch key={location.key}>
        <Route exact path="/" location={location} component={Home} />
        <Route path="/clients" location={location} component={Clients} />
        <Route path="/contacts" location={location} component={Contacts} />
        <Route exact path="/products" location={location} component={Products} />
        <Route path="/products/:category" location={location} component={Products} />
        <Route path="/products/:category" location={location} component={Products} />
        <Route path="/products/:category" location={location} component={Products} />
        <Route component={Home}/>
    </Switch>

    );

export default Routes;
