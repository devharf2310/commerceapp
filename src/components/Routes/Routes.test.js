import React from 'react'
import renderer from 'react-test-renderer'
import Routes from './Routes';
import MockRouter from 'react-mock-router';

const location = { key: ''};
describe('routes snapshot tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MockRouter location={location}><Routes location={location}/></MockRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
