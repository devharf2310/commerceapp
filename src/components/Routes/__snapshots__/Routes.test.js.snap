// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes snapshot tests renders correctly 1`] = `
<h1>
  Home
</h1>
`;
